resources/lib/vrtplayer/tvguide.py

```diff
@@ -0,0 +1,122 @@
+from datetime import datetime, timedelta
+import os
+import requests
+
+from resources.lib.helperobjects import helperobjects
+from resources.lib.vrtplayer import actions, metadatacreator, statichelper
+
+CHANNELS = dict(
+    een=dict(
+        id='O8',
+        name='Eén',
+    ),
+    canvas=dict(
+        id='1H',
+        name='Canvas',
+    ),
+    ketnet=dict(
+        id='O9',
+        name='Ketnet',
+    ),
+)
+
+DATES = {
+    '1': 'Tomorrow',
+    '0': 'Today',
+    '-1': 'Yesterday',
+}
+
+
+class TVGuide:
+
+    VRT_TVGUIDE = 'https://www.vrt.be/bin/epg/schedule.%Y-%m-%d.json'
+
+    def __init__(self, addon_path, kodi_wrapper):
+        self._addon_path = addon_path
+        self._kodi_wrapper = kodi_wrapper
+        self._proxies = self._kodi_wrapper.get_proxies()
+
+    def show_tvguide(self, params):
+        date = params.get('date')
+        channel = params.get('channel')
+
+        if not date:
+            today = datetime.today()
+            date_items = []
+            for i in range(7, -31, -1):
+                day = today + timedelta(days=i)
+                title = day.strftime(self._kodi_wrapper.get_localized_datelong())
+                if str(i) in DATES:
+                    if i == 0:
+                        title = '[COLOR yellow][B]%s[/B], %s[/COLOR]' % (DATES[str(i)], title)
+                    else:
+                        title = '[B]%s[/B], %s' % (DATES[str(i)], title)
+                date_items.append(
+                    helperobjects.TitleItem(title=title,
+                                            url_dict=dict(action=actions.LISTING_TVGUIDE, date=day.strftime('%Y-%m-%d')),
+                                            is_playable=False,
+                                            art_dict=dict(thumb='DefaultYear.png', icon='DefaultYear.png', fanart='DefaultYear.png'),
+                                            video_dict=dict(plot=day.strftime(self._kodi_wrapper.get_localized_datelong()))),
+                )
+            self._kodi_wrapper.show_listing(date_items, sort='unsorted', content_type='files')
+
+        elif not channel:
+            channel_items = []
+            for channel in ('een', 'canvas', 'ketnet'):
+                channel_items.append(
+                    helperobjects.TitleItem(
+                        title=CHANNELS[channel]['name'],
+                        url_dict=dict(action=actions.LISTING_TVGUIDE, date=date, channel=channel),
+                        is_playable=False,
+                        art_dict=dict(thumb=self.__get_media(channel + '.png'), icon='DefaultAddonPVRClient.png', fanart='DefaultAddonPVRClient.png'),
+                        video_dict=dict(plot='TV guide for channel %s' % CHANNELS[channel]['name']),
+                    ),
+                )
+            self._kodi_wrapper.show_listing(channel_items, content_type='files')
+
+        else:
+            dateobj = datetime.strptime(date, '%Y-%m-%d')
```

Comment:
This causes a crash:
```
dateobj = datetime.strptime(date, '%Y-%m-%d')
TypeError: attribute of type 'NoneType' is not callable
```
It's a known Python bug: https://forum.kodi.tv/showthread.php?tid=112916&pid=1212333#pid1212333

Reply:
Oh right, coincidentally I was thinking we should document this better in our code, not realizing I was introducing the same thing some hours later :-(

Let's make this a function call instead...

Reply:
Fixed.